python/lsst/ts/wep/task/calcZernikesNeuralTask.py

```diff
@@ -0,0 +1,1299 @@
+        process any donut data. It handles two scenarios: when there are
+        no donuts at all, and when there are donuts but they all fail
+        quality checks.
+
+        Parameters
+        ----------
+        qualityTable : astropy.table.QTable, optional
+            Quality table created from donut stamp input. If provided, this
+            table will be included in the output even if all donuts failed
+            quality checks. If None, an empty quality table will be created.
+
+        Returns
+        -------
+        lsst.pipe.base.Struct
+            A struct containing empty or failed results:
+            - outputZernikesRaw : np.ndarray
+                Array filled with NaN values for all Noll indices.
+            - outputZernikesAvg : np.ndarray
+                Array filled with NaN values for all Noll indices.
+            - donutStampsNeural : astropy.table.QTable
+                Empty neural network-generated donut stamps collection.
+            - zernikes : astropy.table.QTable
+                Empty Zernike coefficient table.
+            - donutQualityTable : astropy.table.QTable
+                Either the provided quality table or an empty one.
+
+        Notes
+        -----
+        The NaN values in the output arrays indicate that no valid Zernike
+        coefficients could be calculated. The quality table preserves
+        information about why donuts failed, which can be useful for
+        debugging.
+        """
+        qualityTableCols = [
+            "SN",
+            "ENTROPY",
+            "ENTROPY_SELECT",
+            "SN_SELECT",
+            "FINAL_SELECT",
+            "DEFOCAL_TYPE",
+            "FX",
+            "FY",
+        ]
+        self.log.info("Producing empty results; quality table provided: %s", qualityTable is not None)
+        if qualityTable is None:
+            donutQualityTable = QTable({name: [] for name in qualityTableCols})
+        else:
+            donutQualityTable = qualityTable
+
+        # Set stamp attributes to None for empty case
+        self.stampsIntra = None
+        self.stampsExtra = None
+
+        # Create empty Zernike table with metadata
+        emptyZkTable = self.initZkTable()
+        emptyZkTable.meta = self.createZkTableMetadata()
+        # Add empty OOD score column
+        emptyZkTable["ood_score"] = []
+
+        # Create empty donut table matching standard format
+        emptyDonutTable = QTable(
+            {
+                "coord_ra": [],
+                "coord_dec": [],
+                "centroid_x": [],
+                "centroid_y": [],
+                "detector": [],
+                "thx_ccs": [],
+                "thy_ccs": [],
+                "thx_ocs": [],
+                "thy_ocs": [],
+                "th_n": [],
+                "th_w": [],
+                "fx": [],
+                "fy": [],
+                "snr": [],
+                "ood_score": [],
```

Comment:
I think this might be the same as the other list I said should be module level constant?